File: it_support-backend/src/main/java/com/ITsupport/IT/support/App/model/Ticket.java
package com.ITsupport.IT.support.App.model;

import com.ITsupport.IT.support.App.model.enums.EtatTicket;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Ticket {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long ticketId;
    private String description;
    @Enumerated(EnumType.STRING)
    private EtatTicket statut;

    @ManyToOne
    @JoinColumn(name = "utilisateur_id")
    private Utilisateur utilisateur;

    @ManyToOne
    @JoinColumn(name = "technicien_id")
    private Technicien technicien;

    @ManyToOne
    @JoinColumn(name = "equipementId")
    private Equipment equipment;

    @ManyToOne
    @JoinColumn(name = "panneId")
    private Panne panne;


}

File: IT_Support-Angular/src/app/classes/ticket.ts
import {StatutTicket} from "./statut-ticket";
import {EtatTicket} from "../roles";
import {User} from "./user";
import {Equipement} from "./equipement";
import {Panne} from "./panne";
import {Technicien} from "./technicien";

export class Ticket {
  ticketId!:number;
  description!:string;
  etatTicket!:EtatTicket;
  user!:User;
  equipement!:Equipement;
  dateDeCreation!:Date;
  panne!:Panne;
  technicien!:Technicien;
}
